OpenAI_API/ChatFunctions/Function.cs

```diff
@@ -0,0 +1,101 @@
+﻿using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace OpenAI_API.ChatFunctions
+{
+    /// <summary>
+    /// Represents a Function object for the OpenAI API.
+    /// A Function contains information about the function to be called, its description and parameters.
+    /// </summary>
+    /// <remarks>
+    /// The 'Name' property represents the name of the function and must consist of alphanumeric characters, underscores, or dashes, with a maximum length of 64.
+    /// The 'Description' property is an optional field that provides a brief explanation about what the function does.
+    /// The 'Parameters' property describes the parameters that the function accepts, which are represented as a JSON Schema object. 
+    /// Various types of input are acceptable for the 'Parameters' property, such as a JObject, a Dictionary of string and object, an anonymous object, or any other serializable object. 
+    /// If the object is not a JObject, it will be converted into a JObject. 
+    /// Refer to the 'Parameters' property setter for more details.
+    /// Refer to the OpenAI API <see href="https://platform.openai.com/docs/guides/gpt/function-calling">guide</see> and the 
+    /// JSON Schema <see href="https://json-schema.org/understanding-json-schema/">reference</see> for more details on the format of the parameters.
+    /// </remarks>
+    public class Function
+    {
+        /// <summary>
+        /// The name of the function to be called. Must be a-z, A-Z, 0-9, or contain underscores and dashes, with a maximum length of 64.
```

Comment:
Are we providing any validation on this? Can we match it against a regex before sending off to OpenAI?